pkgs/tools/security/iaito/default.nix

```diff
@@ -1,61 +1,65 @@
 { lib
 , stdenv
 , fetchFromGitHub
+, meson
+, ninja
 , pkg-config
+, python3
 , qtbase
 , qttools
 , radare2
 , wrapQtAppsHook
-, zip
-, nix-update-script
 }:
 
+# TODO MacOS support.
+# TODO Build and install translations.
+
 stdenv.mkDerivation rec {
   pname = "iaito";
-  version = "5.7.0";
+  version = "5.7.2";
 
   src = fetchFromGitHub {
     owner = "radareorg";
     repo = pname;
     rev = version;
-    fetchSubmodules = true;
-    sha256 = "sha256-qEJTsS669eEwo2iiuybN72O5oopCaGEkju8+ekjw2zk=";
+    sha256 = "sha256-5/G5wfdc6aua90XLP3B7Ruy8F3NTXzWfQE6yVDZ0rX8=";
   };
 
-  nativeBuildInputs = [ pkg-config qttools wrapQtAppsHook zip ];
+  nativeBuildInputs = [ meson ninja pkg-config python3 qttools wrapQtAppsHook ];
+
   buildInputs = [ radare2 qtbase ];
 
-  postPatch = ''
-    substituteInPlace Makefile \
-      --replace "git submodule update --init" ""
+  postUnpack = ''
+    sourceRoot=$sourceRoot/src
   '';
 
-  NIX_CFLAGS_COMPILE = [ "-I${radare2}/include/libr" "-I${radare2}/include/libr/sdb" ];
+  # TODO Fix version checking and version information for r2.
+  # Version checking always fails due to values being empty strings for some
+  # reason. Meanwhile, we can safely assume that radare2's runtime and
+  # compile-time implementations are the same and remove this check.
+  patches = [ ./remove-broken-version-check.patch ];
 
   installPhase = ''
     runHook preInstall
 
-    install -Dm755 build/iaito $out/bin/iaito
-    install -Dm644 $src/src/org.radare.iaito.appdata.xml $out/share/metainfo/org.radare.iaito.appdata.xml
-    install -Dm644 $src/src/org.radare.iaito.desktop $out/share/applications/org.radare.iaito.desktop
-    install -Dm644 $src/src/img/iaito-o.svg $out/share/pixmaps/iaito-o.svg
+    install -m755 -Dt $out/bin iaito
+    install -m644 -Dt $out/share/applications $src/src/org.radare.iaito.desktop
+    install -m644 -Dt $out/share/pixmaps $src/src/img/iaito-o.svg
+    install -m644 -Dt $out/share/metainfo $src/src/org.radare.iaito.appdata.xml
 
     runHook postInstall
   '';
 
-  passthru.updateScript = nix-update-script {
-    attrPath = pname;
-  };
```

Comment:
Can you re-add this? It's not a bad thing to have one even if r-ryantm doesn't need it. (It works better with one, actually.)

Reply:
r-ryantm and `nix-update` both work fine without it for other packages I maintain. I don't think that there's a real need to keep it here.

Reply:
I (and others) disagree. Specifically, when this was brought up last week, Ryan [said](https://github.com/NixOS/nixpkgs/pull/182549#discussion_r928309115]:

> I think it is still good practice to have an passthru.updateScript because it documents how to update it, gives more control to the maintainers, and is potentially useful for other tools too.

However, I don't think it's very productive to continue going back and forth on this, so I'll drop it.